4.1rc1: Numpy headers not found when using system cfitsio or wcslib

I am trying to prepare the openSUSE package for the next release, [working with 4.1rc1](https://build.opensuse.org/package/show/home:bnavigator:branches:devel:languages:python:numeric/python-astropy-rc).

I am running into compilation errors when trying to unbundle cfitsio and wcslib. Some subpackages do not find the numpy headers. The proper include parameter for gcc is missing:

With `ASTROPY_USE_SYSTEM_CFITSIO=1`:
```
...
[   35s] building 'astropy.io.ascii.cparser' extension
[   35s] creating build/temp.linux-x86_64-3.8/astropy/io
[   35s] creating build/temp.linux-x86_64-3.8/astropy/io/ascii
[   35s] creating build/temp.linux-x86_64-3.8/astropy/io/ascii/src
[   35s] gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -DOPENSSL_LOAD_CONF -fwrapv -fno-semantic-interposition -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -fPIC -I/usr/lib64/python3.8/site-packages/numpy/core/include -I/usr/include/python3.8 -c astropy/io/ascii/cparser.c -o build/temp.linux-x86_64-3.8/astropy/io/ascii/cparser.o
...
[   46s] building 'astropy.io.fits.compression' extension
[   46s] creating build/temp.linux-x86_64-3.8/astropy/io/fits
[   46s] creating build/temp.linux-x86_64-3.8/astropy/io/fits/src
[   46s] gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -DOPENSSL_LOAD_CONF -fwrapv -fno-semantic-interposition -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -fPIC -U_REENTRANT -I/usr/include/cfitsio -I/usr/include/python3.8 -c ./astropy/io/fits/src/compressionmodule.c -o build/temp.linux-x86_64-3.8/./astropy/io/fits/src/compressionmodule.o
[   46s] ./astropy/io/fits/src/compressionmodule.c:100:10: fatal error: numpy/arrayobject.h: No such file or directory
[   46s]   100 | #include <numpy/arrayobject.h>
[   46s]       |          ^~~~~~~~~~~~~~~~~~~~~
[   46s] compilation terminated.
```
Note that  `astropy.io.ascii.cparser`  has the `-I...numpy..` parameter,  `astropy.io.fits.compression` does not.

With `ASTROPY_USE_SYSTEM_WCSLIB=1`:
```
[   10s] building 'astropy.convolution._convolve' extension
[   10s] creating build/temp.linux-x86_64-3.8
[   10s] creating build/temp.linux-x86_64-3.8/astropy
[   10s] creating build/temp.linux-x86_64-3.8/astropy/convolution
[   10s] creating build/temp.linux-x86_64-3.8/astropy/convolution/src
[   10s] gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -DOPENSSL_LOAD_CONF -fwrapv -fno-semantic-interposition -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -fPIC -I/usr/lib64/python3.8/site-packages/numpy/core/include -I/usr/include/python3.8 -c astropy/convolution/src/convolve.c -o build/temp.linux-x86_64-3.8/astropy/convolution/src/convolve.o -UNDEBUG -fPIC
[   10s] gcc -pthread -shared build/temp.linux-x86_64-3.8/astropy/convolution/src/convolve.o -L/usr/lib64 -o build/lib.linux-x86_64-3.8/astropy/convolution/_convolve.cpython-38-x86_64-linux-gnu.so
[   10s] building 'astropy.modeling._projections' extension
[   10s] creating build/temp.linux-x86_64-3.8/astropy/modeling
[   10s] creating build/temp.linux-x86_64-3.8/astropy/modeling/src
[   10s] gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -DOPENSSL_LOAD_CONF -fwrapv -fno-semantic-interposition -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -fPIC -DECHO -DWCSTRIG_MACRO -DASTROPY_WCS_BUILD -D_GNU_SOURCE -DNDEBUG -DHAVE_SINCOS -UDEBUG -I/usr/include/wcslib -Iastropy/modeling/src -I/usr/include/python3.8 -c astropy/modeling/src/projections.c -o build/temp.linux-x86_64-3.8/astropy/modeling/src/projections.o -Wno-strict-prototypes -Wno-unused-function -Wno-unused-value -Wno-uninitialized
[   10s] astropy/modeling/src/projections.c:12:10: fatal error: numpy/arrayobject.h: No such file or directory
[   10s]    12 | #include <numpy/arrayobject.h> /* Numpy */
[   10s]       |          ^~~~~~~~~~~~~~~~~~~~~
[   10s] compilation terminated.

```
In this case these are the first two extensions to be compiled, the first one has the numpy header include parameter, the second does not.

This does not happen when the bundled cfitsio and wcslib are used.

### System Details
(See the logs from the linked OBS project above for the whole system environment and used package versions)
<!-- Even if you do not think this is necessary, it is useful information for the maintainers.
Please run the following snippet and paste the output below:
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("Numpy", numpy.__version__)
import astropy; print("astropy", astropy.__version__)
import scipy; print("Scipy", scipy.__version__)
import matplotlib; print("Matplotlib", matplotlib.__version__)
-->
```
[    4s] Linux-5.7.5-1-default-x86_64-with-glibc2.2.5
[    4s] Python 3.8.3 (default, May 17 2020, 14:48:56) [GCC]
[    4s] Numpy 1.18.4
[    4s] astropy n/a
[    4s] Scipy n/a
[    4s] Matplotlib n/a
```

--- a/astropy/io/fits/setup_package.py
+++ b/astropy/io/fits/setup_package.py
@@ -23,7 +23,8 @@ def _get_compression_extension():
 
     if (int(os.environ.get('ASTROPY_USE_SYSTEM_CFITSIO', 0)) or
             int(os.environ.get('ASTROPY_USE_SYSTEM_ALL', 0))):
-        cfg.update(pkg_config(['cfitsio'], ['cfitsio']))
+        for k, v in pkg_config(['cfitsio'], ['cfitsio']).items():
+            cfg[k].extend(v)
     else:
         if get_compiler() == 'msvc':
             # These come from the CFITSIO vcc makefile, except the last

--- a/astropy/utils/xml/setup_package.py
+++ b/astropy/utils/xml/setup_package.py
@@ -18,7 +18,8 @@ def get_extensions(build_type='release'):
 
     if (int(os.environ.get('ASTROPY_USE_SYSTEM_EXPAT', 0)) or
             int(os.environ.get('ASTROPY_USE_SYSTEM_ALL', 0))):
-        cfg.update(pkg_config(['expat'], ['expat']))
+        for k, v in pkg_config(['expat'], ['expat']).items():
+            cfg[k].extend(v)
     else:
         EXPAT_DIR = 'cextern/expat/lib'
         cfg['sources'].extend([

--- a/astropy/wcs/setup_package.py
+++ b/astropy/wcs/setup_package.py
@@ -191,7 +191,8 @@ def get_wcslib_cfg(cfg, wcslib_files, include_paths):
         wcsconfig_h_path = join(WCSROOT, 'include', 'wcsconfig.h')
         if os.path.exists(wcsconfig_h_path):
             os.unlink(wcsconfig_h_path)
-        cfg.update(pkg_config(['wcslib'], ['wcs']))
+        for k, v in pkg_config(['wcslib'], ['wcs']).items():
+            cfg[k].extend(v)
     else:
         write_wcsconfig_h(include_paths)
 

